docs/design/TECHNICAL_DOCUMENTATION.md

```diff
@@ -0,0 +1,142 @@
+
+### Global architecture of the Parser and Renderer
+
+The Design Token code generator tool is technically split in two main features :
+
+- The **parsers** that will parse and decode the Design Token Specification.
+- The **renderers** that will generate the output (mainly code) based on the decoded Design Token Specification.
+
+Both of these features can be customized at different levels. They can be customized thanks to parameters exposed by parsers and renderers **factory functions**, or they can be completely re-implemented and injected into the Design Token Code generator.
+
+### Parsers
+
+Currently the [o3r/design](https://github.com/AmadeusITGroup/otter/tree/main/packages/@o3r/design) package provides only a single [parser](https://github.com/AmadeusITGroup/otter/tree/main/packages/@o3r/design/src/core/design-token/parsers).
+This parser handle the Design Token Specification as JSON file (or loaded object resulting of a parsing of the JSON file) and result to an object facilitating the tree navigation and the final value calculation to the renderers.
+
+### Renderers
+
+The package is exposing 3 renderers:
+
+- **CSS** Renderer generating CSS Variables according to the specification *(default renderer)*
+- **Sass** Renderer generating Sass Variables according to the specification
+- **Metadata** renderer that will create the metadata file compatible with the Otter CMS solution.
+
+### Example of Code Generator usage
+
+#### Example: Basic renderer usage
+
+```typescript
+import { parseDesignTokenFile, renderDesignTokens } from '@o3r/design';
+
+(async () => {
+
+  /** List of Design Token item parsed */
+  const parsedTokenDesign = await parseDesignTokenFile('./path/to/spec.json');
+
+  // Render the CSS Variable
+  await renderDesignTokens(parsedTokenDesign, { logger: console });
+
+})();
+```
+
+##### Example: CSS Renderer with Sass Fallback
+
+```typescript
+import { getCssTokenDefinitionRenderer, getSassTokenDefinitionRenderer, parseDesignTokenFile, renderDesignTokens } from '@o3r/design';
+
+(async () => {
+
+  /** List of Design Token item parsed */
+  const parsedTokenDesign = await parseDesignTokenFile('./path/to/spec.json');
+
+  /** Sass Variable Renderer */
+  const sassTokenDefinitionRenderer = getSassTokenDefinitionRenderer();
+
+  /** CSS Variable Renderer */
+  const cssTokenDefinitionRenderer = getCssTokenDefinitionRenderer({
+    // Specify that the private variable should be rendered in Sass variable
+    privateDefinitionRenderer: sassTokenDefinitionRenderer
+  });
+
+  // Render the CSS Variable
+  await renderDesignTokens(parsedTokenDesign, { tokenDefinitionRenderer: cssTokenDefinitionRenderer });
+
+})();
+```
+
+##### Example: Implement a Less Renderer
+
+```typescript
+import { getCssTokenDefinitionRenderer, getSassTokenDefinitionRenderer, parseDesignTokenFile, renderDesignTokens } from '@o3r/design';
+
+/** Option of the Less variable renderer */
```

Comment:
```suggestion
/** Options of the Less variable renderer */
```
I know we need a typedoc for all public interface but it is a bit "it does what it says" :P